File: student-project/backend-csharp/StudentApi/Models/Student.cs
// Models/Student.cs
// Entity representing a student in the database

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StudentApi.Models;

[Table("students")]
public class Student
{
    // Primary key - auto-incremented
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    // First name - required, max 100 chars
    [Required(ErrorMessage = "First name is required")]
    [MaxLength(100, ErrorMessage = "First name must be 100 characters or fewer")]
    [Column("first_name")]
    public string FirstName { get; set; } = string.Empty;

    // Last name - required, max 100 chars
    [Required(ErrorMessage = "Last name is required")]
    [MaxLength(100, ErrorMessage = "Last name must be 100 characters or fewer")]
    [Column("last_name")]
    public string LastName { get; set; } = string.Empty;

    // Email - required, unique, max 255 chars
    [Required(ErrorMessage = "Email is required")]
    [EmailAddress(ErrorMessage = "Email must be a valid email address")]
    [MaxLength(255)]
    public string Email { get; set; } = string.Empty;

    // Phone - required, max 20 chars
    [Required(ErrorMessage = "Phone is required")]
    [MaxLength(20, ErrorMessage = "Phone must be 20 characters or fewer")]
    public string Phone { get; set; } = string.Empty;

    // Enrollment date - required
    [Required(ErrorMessage = "Enrollment date is required")]
    [Column("enrollment_date")]
    public DateTime EnrollmentDate { get; set; }

    // Timestamps - managed automatically
    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: student-project/backend-springboot/src/main/java/com/efrei/studentapi/model/Student.java
// Student.java
// JPA entity representing a student in the database

package com.efrei.studentapi.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(
    name = "students",
    indexes = {
        @Index(name = "idx_email", columnList = "email"),
        @Index(name = "idx_enrollment_date", columnList = "enrollment_date")
    }
)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "First name is required")
    @Size(max = 100, message = "First name must be 100 characters or fewer")
    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @NotBlank(message = "Last name is required")
    @Size(max = 100, message = "Last name must be 100 characters or fewer")
    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @NotBlank(message = "Email is required")
    @Email(message = "Email must be a valid email address")
    @Column(nullable = false, unique = true, length = 255)
    private String email;

    @NotBlank(message = "Phone is required")
    @Size(max = 20, message = "Phone must be 20 characters or fewer")
    @Column(nullable = false, length = 20)
    private String phone;

    @NotNull(message = "Enrollment date is required")
    @Column(name = "enrollment_date", nullable = false)
    private LocalDate enrollmentDate;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
